src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/QueueHealthDecider.java

```diff
@@ -92,13 +108,22 @@ private void configureActionPriority() {
    * to consume better signals going forward.
    */
   private Action computeBestAction(NodeKey esNode, ResourceEnum threadPool) {
+    final NodeConfigCache nodeConfigCache = getAppContext().getNodeConfigCache();
+    double heapUsedPercent = HeapUtil.getHeapUsedPercentage(
+            NodeConfigCacheReaderUtil.readHeapMaxSizeInBytes(nodeConfigCache, esNode),
+            NodeConfigCacheReaderUtil.readHeapUsageInBytes(nodeConfigCache, esNode));
     Action action = null;
 
-    for (String actionName : actionsByUserPriority) {
-      action =
-        getAction(actionName, esNode, threadPool, true);
-      if (action != null) {
-        break;
+    // there is no difference in the action when state is UNDER_SIZED or HEALTHY_WITH_BUFFER.
+    // TODO: Tune action parameters based on the state.
+    if (heapUsageBucketCalculator.compute(heapUsedPercent).equals(UsageBucket.HEALTHY_WITH_BUFFER)
```

Comment:
Instead of calling it twice, `heapUsageBucketCalculator.compute(heapUsedPercent)` should we store it in a local variable ?